nemo_rl/algorithms/dpo.py

```diff
@@ -234,12 +233,33 @@ def setup(
     )
 
 
-def add_ref_logprobs_to_data(dataloader, policy, master_config, is_val=False):
+def add_ref_logprobs_to_data(dataloader, policy, master_config, tokenizer, is_val=False):
     dataloader_iter = iter(dataloader)
     while True:
         try:
             batch = next(dataloader_iter)
 
+            ## add loss mask based on role to every message
+            add_loss_mask_to_message_log(
```

Comment:
It seems cleaner (and potentially more efficient) to me to keep all this data preprocessing stuff (including the other lines below) in the dataloader, through the collate function.

Why did you move it?

Reply:
Looks good, thanks